lib/source_opam_config.ml

```diff
@@ -0,0 +1,229 @@
+open Import
+
+module Opam_repositories = struct
+  type t = OpamUrl.Set.t
+
+  let opam_monorepo_cwd_var = "$OPAM_MONOREPO_CWD"
+
+  let opam_monorepo_cwd_var_regexp =
+    let open Re in
+    compile (seq [ str "file://"; str opam_monorepo_cwd_var ])
+
+  let rewrite_url_in ~pos ~opam_monorepo_cwd url_str =
```

Comment:
I think that this rewrite should be a bit more strict. If `$OPAM_MONOREPO_CWD` is not part of the URL, there might be security implications since opam-monorepo can read anywhere in the filesystem.